modules/apps/redirect/redirect-api/src/main/java/com/liferay/redirect/configuration/RedirectPatternConfigurationProvider.java

```diff
@@ -27,7 +27,7 @@
 @ProviderType
 public interface RedirectPatternConfigurationProvider {
 
-	public Map<Pattern, String> getPatternStrings(long groupId)
+	public Map<RedirectPatternUserAgent, String> getPatternStrings(long groupId)
```

Comment:
I don't think a map makes sense anymore. I used it because (conceptually) this was a set of pairs; now we have a set of triplets. I'd change things so that instead of Map<RPUa, String> we'd have List<RedirectPatternEntry>, where RedirectPatternEntry is a (Pattern, Replacement, UserAgent) triplet.

Reply:
No, no. I just called it replacement for no reason :-D Keep the names as is, just group the three items in the `RedirectPatternEntry` class.